scripts/jenkins/ardana/gerrit/gerrit.py

```diff
@@ -0,0 +1,280 @@
+import json
+import re
+import sys
+
+try:
+    from pygerrit2 import GerritRestAPI, GerritReview, HTTPDigestAuthFromNetrc
+except ImportError:
+    # Only reviewing and merging functionality depends on pygerrit2
+    pass
+
+import requests
+
+GERRIT_URL = 'https://gerrit.suse.provo.cloud'
+
+# If running in a virtualenv, the system certificate will not be available
+if hasattr(sys, 'real_prefix') or \
+   hasattr(sys, 'base_prefix') and sys.base_prefix != sys.prefix:
+    GERRIT_VERIFY = False
+else:
+    GERRIT_VERIFY = True
+
+# We use a more complex regex that matches both formats of Depends-On so that
+# we preserve the order in which they are discovered.
+DEPENDS_ON_RE = re.compile(
+    r"^\W*Depends-On: ("
+    r"((http(s)?:\/\/)?gerrit\.suse\.provo\.cloud\/(\/)?(\#\/c\/)?(\d+).*?)"
+    r"|(I[0-9a-f]{40})"
+    r")\s*$",
+    re.MULTILINE | re.IGNORECASE)
+
+
+class GerritChange:
+    """
+    Holds the state of a gerrit change
+    """
+
+    @staticmethod
+    def _query_gerrit(query):
+        query_url = GERRIT_URL + query
+        print("Running query %s" % query_url)
+        response = requests.get(query_url, verify=GERRIT_VERIFY)
+        print("Got response: %s" % response)
+        return json.loads(response.text.replace(")]}'", ''))
+
+    def __eq__(self, other):
+        return self.id == other.id
+
+    def __hash__(self):
+        return hash(self.id)
+
+    def __init__(self, change_id=None, branch=None,
+                 patchset=None, change_object=None):
+        if change_id:
+            print("Processing given change id: %s" % change_id)
+            if change_id.isdigit():
+                self._get_numeric_change(change_id, patchset)
+            elif change_id.split('/')[0].isdigit():
+                change_id, patchset = change_id.split('/')
+                self._get_numeric_change(change_id, patchset)
+            elif change_id.startswith('I') and len(change_id) == 41:
+                self._get_change_id(change_id, branch, patchset)
+            else:
+                raise Exception("Unknown change id format (%s)" % change_id)
+        else:
+            self._set_change_object(change_object)
```

Comment:
We should check that change_object is supplied rather than assuming it is and raise an error if not.